template.rb

```diff
@@ -90,8 +90,15 @@ def copy_solidus_starter_frontend_files
   copy_file 'config/initializers/canonical_rails.rb'
 
   application <<~RUBY
-    # Load monkey patches
-    Rails.autoloaders.main.ignore(Rails.root.join('app/monkey_patches'))
+    if Rails.autoloaders.main
+      # Rails 7
+      Rails.autoloaders.main.ignore(Rails.root.join('app/monkey_patches'))
+    else
+      # Rails 6 and older
+      config.after_initialize do
+        Rails.autoloaders.main.ignore(Rails.root.join('app/monkey_patches'))
+      end
+    end
```

Comment:
nit: Actually here we could even keep the if outside of `application.rb` since the template ran on an existing app with a definite Rails version:

```rb
if Rails.autoloaders.main # Or `Rails.gem_version >= Gem::Version.new('7')`
  application <<~RUBY
    # Load monkey patches
    Rails.autoloaders.main.ignore(Rails.root.join('app/monkey_patches'))
  RUBY
else
  application <<~RUBY
    # Load monkey patches
    config.after_initialize do
      Rails.autoloaders.main.ignore(Rails.root.join('app/monkey_patches'))
    end
  RUBY
end

Reply:
Yeah, this looks better than having that conditional written on the `application.rb` file :+1: